api/types/src/transaction.rs

```diff
@@ -431,64 +445,84 @@ impl TryFrom<Script> for ScriptPayload {
     }
 }
 
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Object)]
 pub struct WriteSetPayload {
     pub write_set: WriteSet,
 }
 
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Union)]
+#[oai(discriminator_name = "type")]
 #[serde(tag = "type", rename_all = "snake_case")]
 pub enum WriteSet {
     ScriptWriteSet(ScriptWriteSet),
     DirectWriteSet(DirectWriteSet),
 }
 
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Object)]
 pub struct ScriptWriteSet {
     pub execute_as: Address,
     pub script: ScriptPayload,
 }
 
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Object)]
 pub struct DirectWriteSet {
     pub changes: Vec<WriteSetChange>,
     pub events: Vec<Event>,
 }
 
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Union)]
+#[oai(discriminator_name = "type")]
 #[serde(tag = "type", rename_all = "snake_case")]
 pub enum WriteSetChange {
-    DeleteModule {
-        address: Address,
-        state_key_hash: String,
-        module: MoveModuleId,
-    },
-    DeleteResource {
-        address: Address,
-        state_key_hash: String,
-        resource: MoveStructTag,
-    },
-    DeleteTableItem {
-        state_key_hash: String,
-        handle: HexEncodedBytes,
-        key: HexEncodedBytes,
-    },
-    WriteModule {
-        address: Address,
-        state_key_hash: String,
-        data: MoveModuleBytecode,
-    },
-    WriteResource {
-        address: Address,
-        state_key_hash: String,
-        data: MoveResource,
-    },
-    WriteTableItem {
-        state_key_hash: String,
-        handle: HexEncodedBytes,
-        key: HexEncodedBytes,
-        value: HexEncodedBytes,
-    },
+    DeleteModule(DeleteModule),
+    DeleteResource(DeleteResource),
+    DeleteTableItem(DeleteTableItem),
+    WriteModule(WriteModule),
+    WriteResource(WriteResource),
+    WriteTableItem(WriteTableItem),
+}
```

Comment:
So all of these changes, change how they serialize right?  So it's a breaking change to the REST API?

Reply:
It doesn't change the structure of the serialization, but there are other breaking changes, e.g. around data format. For example, the new API tries to format numbers as numbers, not strings. Also some structs are unpacked, e.g. this kind of thing:

```
curl http://localhost:8080/accounts/0x1/resources | jq .[0]
{
  "type": "0x1::Coin::CoinInfo<0x1::TestCoin::TestCoin>",
  "data": {
    "decimals": "6",
    "name": "Test Coin",
    "supply": {
      "vec": []
    },
    "symbol": "TC"
  }
}

curl http://localhost:8080/v1/accounts/0x1/resources | jq .[0]
{
    "type": {
      "address": "0x1",
      "module": "Coin",
      "name": "CoinInfo",
      "generic_type_params": [
        "0x1::TestCoin::TestCoin"
      ]
    },
    "data": {
      "decimals": "6",
      "name": "Test Coin",
      "supply": {
        "vec": []
      },
      "symbol": "TC"
    }
}
```